src/ServiceStack.Mvc/ServiceStackController.cs

```diff
@@ -53,7 +53,17 @@ public virtual ActionResult AuthenticationErrorResult
             get
             {
                 var returnUrl = HttpContext.Request.GetPathAndQuery();
-                return new RedirectResult(UnauthorizedRedirectUrl.Fmt(returnUrl.UrlEncode()));
+
+                var encoded = returnUrl.UrlEncode();
+                var unauthorizedRedirectUrl = UnauthorizedRedirectUrl;
+
+                if (string.IsNullOrEmpty(unauthorizedRedirectUrl))
+                {
+                    throw new HttpException(401, "Unauthorized");
+                }
+
+                var formatted = unauthorizedRedirectUrl.Fmt(encoded);
```

Comment:
Ok, I did raise this as an issue first, as I wasn't sure.

The main point is, in the documentation it says that if you set HtmlRedirect to null, you will get a 401 response when unauthorised, instead you get a 302, which I don't want.

I only want a 302 when I set HtmlRedirect to an actual value, with the temporary redirection being to that value.